### ISSUE TITLE
Add feature to move ROM from data slot to extsd slot

### ISSUE THREAD

**Issue by webmaster33** [chenxiaolong/DualBootPatcher#371]
Would be fine to have a feature to move ROM from data slot to extsd slot. 

The reason is, that flashing to data slot is much faster than installing directly to the extsd slot. 

When an OS is successfully installed to data slot, then could be possible to move to extsd slot to free up the space on internal memory.

**Comment by wazeeahmed** [comment]
Well , in that case you can backup whole rom and restore to any slot you want . This feature likely unnecessary for now .

**Comment by webmaster33** [comment]
Not unnecessary.
The data slot flashing takes only about 20-30 minutes, while extsd flashing takes about 6-7 hours.

So flashing to data slot is much faster.

**Comment by wazeeahmed** [comment]
Then whats the problem ? You already have it in data slot don't you ? Maybe a higher class sd card like sdxc uhs 1 or 2 class will fix that time taking problem for ya .

**Comment by webmaster33** [comment]
I use class 10 Samsung EVO 64GB sdcard.

**Comment by wazeeahmed** [comment]
Have you checked the sd card speed ? maybe the sd card has some issues . I use samsung sdhc class 10 16gb and here it's just a few minutes more than regular rom flashing time . But the difference is hardly noticeable . Yours should do much beter in this case . Check your card for performance issues .

**Comment by webmaster33** [comment]
It is a common problem, for Cyanogenmod roms, that at message "patching system rom unconditionally" takes unusually long time. 
Strangely this doesn't happen at flashing data slot.

**Comment by Kingzak34** [comment]
And anyway the time to transfer from internal to external would be the same as the time to flash directly to external.

**Comment by webmaster33** [comment]
I checked the card with USB 3.0 card reader, reformatted with exFAT. 
The performance was as I remember about 40 Mb/s read and about 20 Mb/s write.
There is no performance problem with the card.

**Comment by webmaster33** [comment]
No, transferring a 3Gb file from internal to external does not take 6-7 hours.

**Comment by Kingzak34** [comment]
A system FS is not a simple 3Gb file.

**Comment by webmaster33** [comment]
system.img file is a simple file.

**Comment by Kingzak34** [comment]
When you install rom in data partition it's unpacked FS. Not a system.img.

**Comment by Kingzak34** [comment]
(See here http://www.cjoint.com/c/GAuoOxIz4Wm)

**Comment by wazeeahmed** [comment]
@Kingzak34 is right . unpacked fs and img both are quite different process in dbp . dbp repacks the rom when installed in extsd slot . But in data it doesn't . Thats where time becomes a matter .

**Comment by webmaster33** [comment]
@wazeeahmed: maybe packing

**Comment by webmaster33** [comment]
@wazeeahmed: maybe packing takes more time, but I think that packing would take 6-7 hours. 

Are there any other Samsung Galaxy S4 i9505 users, who are flashing Cyanogenmod 13 to extsd slot, and takes 6-7 hours?
